tests/integration/autoTracking.spec.js

```diff
@@ -0,0 +1,770 @@
+ *   derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import util from 'util'
+import F from 'lodash/fp'
+import { fetchResults, start, stop } from '../micro'
+
+const dumpLog = log => console.log(util.inspect(log, true, null, true))
+
+const isMatchWithCallback = F.isMatchWith((lt, rt) =>
+  F.isFunction(rt) ? rt(lt) : undefined
+)
+
+const SAFARI_EXPECTED_FIRST_NAME = 'Alex'
+const SAFARI_EXPECTED_MESSAGE = 'Changed message'
+
+describe('Auto tracking', () => {
+  if (F.isMatch({ browserName: 'internet explorer', version: '9' }, browser.capabilities)) {
+    fit('Skip IE9', () => {}) // Automated tests for IE autotracking features 
+  }
+
+  let log = []
+  let docker
+
+  const logContains = ev => F.some(isMatchWithCallback(ev), log)
+
+  const loadUrlAndWait = (url) => {
+    browser.url(url)
+    browser.waitUntil(
+      () => $('#init').getText() === 'true',
+      {
+        timeout: 5000,
+        timeoutMsg: 'expected init after 5s',
+        interval: 250
+      }
+    )
+  }
+
+  beforeAll(() => {
+    browser.call(() => {
+      return start()
+        .then((container) => {
+          docker = container
+        })
+    })
+    browser.url('/index.html')
+    browser.setCookies({ name: 'container', value: docker.url })
+  })
+
+  afterAll(() => {
+    browser.call(() => {
+      return stop(docker.container)
+    })
+  })
+
+  it('should send a link click events', () => {
+    loadUrlAndWait('/link-tracking.html')
+    $('#link-to-click').click()
+    
+    loadUrlAndWait('/link-tracking.html?filter=exclude')
+    $('#link-to-not-track').click()
+    $('#link-to-click').click()
+
+    loadUrlAndWait('/link-tracking.html?filter=filter')
+    $('#link-to-filter').click()
+    $('#link-to-click').click()
+
+    loadUrlAndWait('/link-tracking.html?filter=include')
+    $('#link-to-filter').click()
+    $('#link-to-click').click()
+
+    // time for activity to register and request to arrive
+    browser.pause(5000)
+    browser.call(() =>
+      fetchResults(docker.url).then(result => {
+        log = result
+      })
+    )
+  })
+
+  it('should send a link click event', () => {
```

Comment:
Perhaps we should also check that the `page_url` field is as expected on the link click event?